ensembl/production/reporting/amqp_reporter.py

```diff
@@ -79,21 +99,21 @@ def smtp_reporter():
     def on_message(message: Message):
         logger.debug("From queue: %s, received: %s", config.amqp_queue, message.body)
         try:
-            email = json.loads(message.body)
-        except json.JSONDecodeError as err:
-            logger.critical("Cannot decode JSON message: %s", message.body)
+            email = validate_payload(message.body)
+            msg = compose_email(email)
+        except ValueError as err:
+            logger.error("%s Email Message: %s", err, email)
             message.reject()
-            logger.debug("Rejected: %s", message.body)
+            logger.warning("Rejected: %s", message.body)
             return
-        msg = compose_email(email)
         try:
             smtp.send_message(msg)
         except SMTPException as err:
-            logger.critical("Unable to send email message: %s", message.body)
+            logger.error("Unable to send email message. %s Message: %s", err, email)
             message.requeue()
```

Comment:
Will message get requeued like forever in case mail is not sent? Is it something intended?

Reply:
Yes. `requeue()` here is to prevent the message to be lost if the SMTP is down. In order to prevent a infinite loop in case the message cannot be sent for other reasons, we should set a TTL policy on the queue and an additional "dead-letter" queue in which to place messages causing problems (i.e. expired and rejected ones) which can then be treated separately.

See: https://www.rabbitmq.com/ttl.html

Reply:
Nice. Thanks for the details.